dataprofiler/tests/profilers/profiler_options/test_histogram_option.py

```diff
@@ -0,0 +1,125 @@
+            option._set_helper({'bin_count_or_method.is_enabled': True}, 'test')
+
+    def test_set(self):
+        option = self.get_options()
+
+        params_to_check = [
+            dict(prop='is_enabled', value_list=[False, True]),
+            dict(prop='bin_count_or_method',
+                 value_list=[None, 'auto', 'fd', 'doane', 'scott', 'rice',
+                             'sturges', 'sqrt', ['sturges', 'doane'], 1,
+                             10, 100, 1000, 99, 10000000])
+        ]
+
+        # this code can be abstracted to limit code everywhere else
+        # AKA, params_to_check would be the only needed code plus raise errors
+        def _assert_set_helper(prop, value):
+            option.set({prop: value})
+            self.assertEqual(value, getattr(option, prop), msg=prop)
+
+        for params in params_to_check:
+            prop, value_list = params['prop'], params['value_list']
+            for value in value_list:
+                _assert_set_helper(prop, value)
+
+        # Treat bin_count_or_method as a BooleanOption
+        expected_error = ("type object 'bin_count_or_method' has no attribute "
+                          "'is_enabled'")
+        with self.assertRaisesRegex(AttributeError, expected_error):
+            option.set({'bin_count_or_method.is_enabled': True})
+
+    def test_validate_helper(self):
+        super(TestHistogramOption, self).test_validate_helper()
+
+    def test_validate(self):
+
+        super(TestHistogramOption, self).test_validate()
+
+        params_to_check = [
+            # non errors
+            dict(prop='is_enabled', value_list=[False, True], errors=[]),
+            dict(prop='bin_count_or_method',
+                 value_list=['auto', 'fd', 'doane', 'scott', 'rice', 'sturges',
+                             'sqrt', ['sturges', 'doane'], 1, 10, 100, 1000,
+                             99, 10000000],
+                 errors=[]),
+
+            # errors
+            dict(prop='bin_count_or_method',
+                 value_list=[-1, 1.2, 1.0, [], False],
+                 errors=[
+                     "HistogramOption.bin_count_or_method must be an integer "
+                     "more than 1, a string, or list of strings from the "
+                     "following: ['auto', 'fd', 'doane', 'scott', 'rice', "
+                     "'sturges', 'sqrt']."
+                 ]),
+            dict(prop='bin_count_or_method',
+                 value_list=['whoops', ["doane", "incorrect"]],
+                 errors=[
+                     r"HistogramOption.bin_count_or_method must be a subset or "
+                     r"selection from: ['auto', 'fd', 'doane', 'scott', 'rice', "
+                     r"'sturges', 'sqrt']."
+                 ]),
+        ]
+
+        # # this code can be abstracted to limit code everywhere else
+        # # AKA, params_to_check would be the only needed code
+        # def _assert_validate_helper(prop, value):
+        #
```

Comment:
Should this get deleted?

Reply:
no bc the code below could be abstracted, but updated the comment